### PR TITLE
[SW-18] Send modal

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

**Comment by CedrikNikita** [comment]
Except of additional commits
Introduced additional changes in this PR on top of what already have been done:
- add logic for every type of send action
- **fix**: deeplinks to `/tips`
- **fix**: openwith funcitonality (sharing via wallet)
- **fix**: validation of fields
- **fix**: disabling of next button
- **fix**: retip page
- **fix**: prefill send modal if selectedToken prop is not null
- **fix**: success send modal being full screen instead of fitting the app width, height
- **fix**: showing nan as AE price if balance is 0
- **fix(name-item)**: show name properly on (error condition was broken)
- **refactor(token-details)**: remove unused code
- **refactor(send)**: remove unused code
- **fix(helpers)**: escapeSpecialChars default value
- **test**: make them work

File: src/popup/router/components/Modals/ConfirmTip.vue [hunk 1]
```diff
@@ -3,12 +3,13 @@
     has-close-button
     @close="resolve"
   >
-    <h2 class="text-heading-2 text-center confirmation-title">
+    <!--     TODO - pawel - remove probably -->
```
Comment (pstachel) [inline src/popup/router/components/Modals/ConfirmTip.vue:6]: can be removed :)

File: src/popup/router/components/Modals/RecipientHelper.vue [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  >
+    <template #msg>
+      <div
+        class="msg"
+      >
+        {{ msg }}
+        <!-- temporarily disabled as the link is down -->
+        <p
+          v-show="false"
+        >
```
Comment (pstachel) [inline src/popup/router/components/Modals/RecipientHelper.vue:16]: Maybe we should rely on UNFINISHED_FEATURES flag here?

File: src/popup/router/components/Modals/RecipientInfo.vue [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            tag="div"
+          >
+            <strong class="title">
+              {{ $t('modals.recipient.msg.url.title') }}:
+            </strong>
+          </i18n>
+        </p>
+
+        <!-- temporarily disabled as the link is down -->
+        <p v-if="false">
```
Comment (pstachel) [inline src/popup/router/components/Modals/RecipientInfo.vue:48]: Maybe we should rely on UNFINISHED_FEATURES flag here?

File: src/popup/router/components/InputAmount.vue [hunk 4]
```diff
@@ -5,23 +5,20 @@
...
     type="number"
     v-bind="$attrs"
     placeholder="0.00"
     :label="$attrs.label || $t('pages.tipPage.amountLabel')"
     @input="$emit('input', $event)"
   >
-    <template slot="right">
-      <span class="token">{{ selectedToken ? selectedToken.symbol : 'AE' }}</span>
+    <template #after>
+      <span class="token">{{ 'AE' }}</span>
```
Comment (peronczyk) [inline src/popup/router/components/InputAmount.vue:20]:
```suggestion
      <span class="token">AE</span>
```

No need to use moustache here.

File: src/popup/router/components/TransferSendForm.vue [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+    border-radius: 12px;
+
+    &:hover {
+      background: rgba(17, 97, 254, 0.15);
+    }
+
+    &.chosen {
+      background: rgba(17, 97, 254, 0.15);
+      border-color: rgba(17, 97, 254, 0.5);
+    }
```
Comment (pstachel) [inline src/popup/router/components/TransferSendForm.vue:366]: Those color should come from variables

File: src/store/plugins/pendingTransactionHandler.js [hunk 6]
```diff
@@ -1,18 +1,26 @@
+import { MODAL_SPEND_SUCCESS } from '../../popup/utils/constants';
+
 export default async (store) => {
   const waitTransactionMined = async ({
     hash, type, amount, tipUrl, recipient: recipientId,
   }) => {
     const network = store.getters.activeNetwork?.networkId;
     try {
       const transaction = await store.state.sdk.poll(hash);
+      // TODO - pawel - close tx modal
```
Comment (pstachel) [inline src/store/plugins/pendingTransactionHandler.js:10]: This TODO should be removed

**Review by pstachel — CHANGES_REQUESTED** [review]

File: src/popup/router/components/InputField.vue [hunk 7]
```diff
@@ -1,13 +1,13 @@
 <template>
   <div
     class="input-field"
-    :class="{
+    :class="[{
```
Comment (peronczyk) [inline src/popup/router/components/InputField.vue:4]: No need to wrap the object with an array.

File: src/popup/router/components/Modals/ConfirmTip.vue (hunk 1 above)
Comment (peronczyk) [inline src/popup/router/components/Modals/ConfirmTip.vue:6]: @pstachel ?? :)

File: src/popup/router/components/Modals/ConfirmTip.vue [hunk 8]
```diff
@@ -3,12 +3,13 @@
...
+    <template slot="header">
```
Comment (peronczyk) [inline src/popup/router/components/Modals/ConfirmTip.vue:7]: `#header`

File: src/popup/router/components/Modals/ConfirmTip.vue [hunk 9]
```diff
@@ -3,12 +3,13 @@
...
       {{ $t('modals.confirm-tip.title') }}
-    </h2>
+    </template>
 
-    <p>{{ $t('modals.confirm-tip.msg') }}</p>
-    <p class="confirmation-question">
+    {{ $t('modals.confirm-tip.msg') }}
+    <p class="confirmation--question">
```
Comment (peronczyk) [inline src/popup/router/components/Modals/ConfirmTip.vue:12]: I would be really happy to use BEM, but we are not using it currently. So I think one dash should be used.

File: src/popup/router/components/Modals/QrCodeReader.vue [hunk 10]
```diff
@@ -146,7 +163,8 @@ export default {
             document.querySelector(s).style = 'background: transparent';
           });
           document.querySelector('.main').style.display = 'none';
-          this.$store.commit('setPageTitle', 'Scan QR');
+          document.querySelector('.transfer-send-modal').style.display = 'none'; // TODO: refactor cordova qr code reading
```
Comment (CedrikNikita) [inline src/popup/router/components/Modals/QrCodeReader.vue:166]: It's a temporary decision, we should move it to the issues after merge.

File: src/popup/router/components/AccountRow.vue [hunk 11]
```diff
@@ -0,0 +1,88 @@
+<template>
+  <div class="account-row">
```
Comment (CedrikNikita) [inline src/popup/router/components/AccountRow.vue:2]: The name doesn't sound perfect for me, feel free to propose a different naming.

**Pull request merged**